File: iPhone/VerBank-CSTSv3-IOSClientApi/ClientStation/api/event/API_IDEvent.m
//
//  API_IDEvent.m
//  JEDIv7-CSTSv3-ClientAPI
//
//  Created by baolin on 13/5/14.
//  Copyright (c) 2013年 wangyubo. All rights reserved.
//

#import "API_IDEvent.h"

@implementation API_IDEvent

-(id)init{
    if(self=[super init]){
        return [self init:@""];
    }
    return self;
}

-(id)init:(NSString *)_eventName{
    if(self=[super init]){
        return [self init:_eventName eventData:nil eventResult:true eventSourceID:@""];
    }
    return self;
}
-(id)init:(NSString *)_eventName eventData:(NSMutableArray *)_eventData  eventResult:(Boolean)_eventResult  eventSourceID:(NSString *)_eventSourceID{
    if (self=[super init]) {
        eventSourceID=_eventSourceID;
        eventName=_eventName;
        eventData=_eventData;
        eventResult=_eventResult;
    }
    return self;
}
 
-(Boolean)isEventResult{
    return eventResult;
}

-(void)setEventResult:(Boolean)_eventResult{
    eventResult=_eventResult;
}

-(NSString *)getEventSourceID{
    return eventSourceID;
}

-(void)setEventSourceID:(NSString *)_eventSourceID{
    eventSourceID=_eventSourceID;
}

-(NSString *)getEventName{
    return eventName;
}

-(void)setEventName:(NSString *)_eventName{
    eventName=_eventName;
}

-(NSMutableArray *)getEventData{
    return eventData;
}

-(void)setEventData:(NSMutableArray *)_eventData{
    eventData=_eventData;
}
@end

File: WeChatService/EX02-CSTSv3-ClientAPI/src/jedi/ex01/client/station/api/event/API_IDEvent.java
package jedi.ex01.client.station.api.event;

public class API_IDEvent implements API_IDEvent_NameInterface {
	private String eventSourceID = "";
	private String eventName;
	private Object[] eventData;
	private boolean eventResult = true;

	public API_IDEvent() {
		this("");
	}

	public API_IDEvent(String eventName) {
		this(eventName, null, true, "");
	}

	public API_IDEvent(String eventName, Object[] eventData,
			boolean eventResult, String eventSourceID) {
		super();
		this.eventSourceID = eventSourceID;
		this.eventName = eventName;
		this.eventData = eventData;
		this.eventResult = eventResult;
	}

	public boolean isEventResult() {
		return eventResult;
	}

	public void setEventResult(boolean eventResult) {
		this.eventResult = eventResult;
	}

	public String getEventSourceID() {
		return eventSourceID;
	}

	public void setEventSourceID(String eventSourceID) {
		this.eventSourceID = eventSourceID;
	}

	public String getEventName() {
		return eventName;
	}

	public void setEventName(String eventName) {
		this.eventName = eventName;
	}

	public Object[] getEventData() {
		return eventData;
	}

	public void setEventData(Object[] eventData) {
		this.eventData = eventData;
	}
}
